File: front-end/modify-about-image.js
import Jimp from 'jimp';
import { fileURLToPath } from 'url';
import { dirname, join } from 'path';

const __filename = fileURLToPath(import.meta.url);
const __dirname = dirname(__filename);

const inputImage = join(__dirname, 'public', 'images', 'about.png');
const outputImage = join(__dirname, 'public', 'images', 'about.png'); // Overwrite original

// Dark indigo blue color: #1a1a2e
const backgroundColor = Jimp.rgbaToInt(26, 26, 46, 255);

async function modifyImage() {
  try {
    console.log('Loading image:', inputImage);
    
    // Load the image
    const image = await Jimp.read(inputImage);
    
    console.log('Image dimensions:', image.getWidth(), 'x', image.getHeight());
    
    // Create a new image with the background color
    const background = new Jimp(image.getWidth(), image.getHeight(), backgroundColor);
    
    // Composite the original image over the background
    // This replaces transparent areas with the background color
    background.composite(image, 0, 0, {
      mode: Jimp.BLEND_SOURCE_OVER,
      opacitySource: 1,
      opacityDest: 1
    });
    
    // Also replace white/light pixels in the composited image with the background color
    background.scan(0, 0, background.getWidth(), background.getHeight(), function (x, y, idx) {
      const red = this.bitmap.data[idx];
      const green = this.bitmap.data[idx + 1];
      const blue = this.bitmap.data[idx + 2];
      const alpha = this.bitmap.data[idx + 3];
      
      // If pixel is white or very light (and not transparent), replace with background
      if (alpha > 0 && red > 240 && green > 240 && blue > 240) {
        this.bitmap.data[idx] = 26;     // R
        this.bitmap.data[idx + 1] = 26; // G
        this.bitmap.data[idx + 2] = 46; // B
        // Keep original alpha
      }
    });
    
    // Save the modified image
    await background.writeAsync(outputImage);
    
    console.log('Image modified successfully!');
    console.log('Output saved to:', outputImage);
    
  } catch (error) {
    console.error('Error modifying image:', error);
    process.exit(1);
  }
}

modifyImage();

File: front-end/modify-about-image.py
#!/usr/bin/env python3
from PIL import Image
import os

# Dark indigo blue color: #1a1a2e
BACKGROUND_COLOR = (26, 26, 46)

input_image = 'public/images/about.png'
output_image = 'public/images/about.png'  # Overwrite original

try:
    print(f'Loading image: {input_image}')
    
    # Load the image
    img = Image.open(input_image)
    
    # Convert to RGBA if not already
    if img.mode != 'RGBA':
        img = img.convert('RGBA')
    
    print(f'Image dimensions: {img.width} x {img.height}')
    print(f'Image mode: {img.mode}')
    
    # Create a new image with the background color
    background = Image.new('RGBA', img.size, BACKGROUND_COLOR + (255,))
    
    # Composite the original image over the background
    # This replaces transparent areas with the background color
    result = Image.alpha_composite(background, img)
    
    # Also replace white/light pixels with the background color
    pixels = result.load()
    for y in range(result.height):
        for x in range(result.width):
            r, g, b, a = pixels[x, y]
            # If pixel is white or very light (and not transparent), replace with background
            if a > 0 and r > 240 and g > 240 and b > 240:
                pixels[x, y] = BACKGROUND_COLOR + (a,)
    
    # Save the modified image
    result.save(output_image, 'PNG')
    
    print('Image modified successfully!')
    print(f'Output saved to: {output_image}')
    
except Exception as error:
    print(f'Error modifying image: {error}')
    import traceback
    traceback.print_exc()
    exit(1)

